Completed Project

Blevs opened the pull request:
@eeevvaaa

Blevs committed:
Sample post content

Blevs committed:
Create Post component

Blevs committed:
Create post header

Blevs committed:
Add card component

Blevs committed:
Add footer component.

It was necessary to upgrade react-scripts to >2.0.0 to import the svg's as components.

The onclick functionality has not yet been implemented.

Blevs committed:
First pass header css

Blevs committed:
Card styling

Blevs committed:
Footer styling

Blevs committed:
Add state and interaction for footer buttons

Blevs committed:
Add card link

Blevs committed:
Numbers and display working

Blevs committed:
Buttons add

Blevs committed:
Actions work

Blevs committed:
Style calculator

Blevs committed:
Fix bug to allow stack to have multiple numbers

Blevs committed:
Fix stack bug

Blevs committed:
Commit yarn.locks

Blevs committed:
Make app behave more like real calculator

This was done using 'clever' logic based on the existing state variables that
should probably have been refactored into more explicit state variables

Blevs committed:
Add css action states to buttons

Blevs committed:
Make handler functions arrow functions

Blevs committed:
Switch to destructured props

And replace the few unnecessary class components with functional equivalents

Blevs committed:
Split calculator into components

Blevs committed:
Switch handler functions to arrow functions

Blevs committed:
Switch to props destructuring

Blevs committed:
Remove unnecessary import

Blevs committed:
Fix division bug where 0 is dividend

Because 0 is equivalent to false during inside if the stack was incorrectly
being reset when it had a value of 0 and not explicitly false.

Blevs committed:
Fix display overflow

Note that this is harcoded to the calculator max width

eeevvaaa reviewed (approved):
Awesome job Henry! I like that you started implementing index files for each component. That is a great practice for going into Redux. 👍

Blevs committed:
Add decimals

This necessitated changing the state variable `total` to a string and parsing it
into a number when required. This is because we cannot add `.0` to the end of a
number, as numerically it doesn't change the value, only the formatting.

Blevs committed:
Make display monospaced to avoid overflow on wide numbers
